### DISCUSSION TITLE
Is it possible to have create a bulk action button without selecting any records?

### DISCUSSION THREAD

**Discussion by paramwork** [filamentphp/filament#5502] [Help]
I am coming from "[laravel-livewire-tables](https://github.com/rappasoft/laravel-livewire-tables)" package. I like the feature that shows the "Bulk Action" Button on the table header.

![chrome_zXLaRXE4nf](https://user-images.githubusercontent.com/63786245/213539998-23809a7d-632f-4168-8cd3-32c0386fd7ca.png)

For my use case, it helps me to configure actions that do not require users to select any rows. Like "Exporting all records". I know it is possible to choose a record and then click on select all to select every record.

Does anybody know if I can create a button possible with a dropdown that allows me to create custom actions?

**Comment by danharrin** [comment]
You could use a `headerActions()` for this instead of a `bulkActions()` then?

**Reply by paramwork** [reply]
Thanks for answering.

How do I use `headerActions()`? I am using the table package.

My actions look like this.

```
protected function getTableActions(): array
    {
        return [
            Action::make('delete')
                ->action(fn (Submission $record) => $record->delete())
                ->openUrlInNewTab()
                ->icon('heroicon-s-trash')
                ->color('danger')
                ->requiresConfirmation()
                ->modalHeading('Delete submission')
                ->modalSubheading('Are you sure you\'d like to delete this submission? This cannot be undone.')
                ->modalButton('Yes, delete')
                ->tooltip('Delete this record')
        ];
    }
  
protected function getTableBulkActions(): array
    {
        return [
            BulkAction::make('delete')
                ->action(fn (Collection $records) => $records->each->delete())
                ->deselectRecordsAfterCompletion()
                ->icon('heroicon-o-trash')
                ->color('danger')
                ->requiresConfirmation()
        ];
    }

    
```

**Reply by danharrin** [reply]
```php
protected function getTableHeaderActions(): array
    {
        return [
            Action::make('delete')
                ->action(fn () => $this->getTableQuery()->delete())
                ->icon('heroicon-o-trash')
                ->color('danger')
                ->requiresConfirmation()
        ];
    }
```

**Reply by paramwork** [reply]
Thank you it works.

**Comment by aymanalhattami** [comment]
How do I obtain a table query based on  filters from the resource to utilize in header action?

**Reply by danharrin** [reply]
`$this->getFilteredTableQuery()`

**Reply by aymanalhattami** [reply]
How Can I use ```$this->getFilteredTableQuery()``` in the resource class ?

**Reply by danharrin** [reply]
You're in a table header action, so `$livewire->getFilteredTableQuery()`. I assumed you were in a page header action, not the resource class

**Reply by aymanalhattami** [reply]
It works
Thanks @danharrin

**Reply by aymanalhattami** [reply]
I have another question

how can I apply the sort to the query on the resource class?

in Page header action, I can use 
```
$query = $this->getFilteredTableQuery();
$this->applySortingToTableQuery($query)
```

but when I use ```$livewire->applySortingToTableQuery($livewire->getFilteredTableQuery())``` in the resource, I got an error
